src/components/presets/account_preset.cairo

```diff
@@ -96,4 +104,21 @@ pub mod AccountPreset {
             self.lockable.is_locked()
         }
     }
+
+    // *************************************************************************
+    //                              PERMISSIONABLE IMPL
+    // *************************************************************************
+    #[abi(embed_v0)]
+    impl Permissionable of IPermissionable<ContractState> {
+        fn set_permission(
+            ref self: ContractState,
+            permission_addresses: Array<ContractAddress>,
+            permissions: Array<bool>
+        ) {
+            self.permissionable.set_permission(permission_addresses, permissions)
+        }
+        fn has_permission(self: @ContractState, permission_addresses: ContractAddress) -> bool {
+            self.permissionable.has_permission(permission_addresses)
+        }
+    }
```

Comment:
within the `account` component, update the `_is_valid_signer` internal function to call `has_permissions` from the permissionable component, and return true if the given caller or signer has permissions.

To do this, you need to embed the `permissionable` component in the `account` component.

This will ensure a permissioned address can execute calls on behalf of the owner. Ensure to also write tests for this

Reply:
so this mean from `_is_valid_signer` Function

we will check for permission from the ELSE CASE

Reply:
wdym by from the ELSE CASE?

Reply:
inside` _is_valid_signer`

there is a check if ` signer == owner` which return true 

```
if (signer == owner) {
                return true;
            } else {
                // run the chek here 
                return false;
            }
```

Reply:
we want to return true if signer == owner, or if signer has permissions, yes.

Reply:
Alright